e2e/etcd_test.go

```diff
@@ -198,9 +212,72 @@ func newEtcdProcessCluster(cfg *etcdProcessClusterConfig) (*etcdProcessCluster,
 			return nil, err
 		}
 	}
+	if epc.cfg.proxySize > 0 {
+		for i := 0; i < 5; i++ {
+			ok, _ := isProxyReady(epc)
+			if ok {
+				break
+			}
+			time.Sleep(time.Second)
+		}
+	}
 	return epc, nil
 }
 
+func isProxyReady(clus *etcdProcessCluster) (bool, error) {
```

Comment:
I don't understand why this is necessary-- isn't the proxy ready after it says it's listening?